src/CounterInterceptor/MyCounterInterceptor.cs

```diff
@@ -0,0 +1,137 @@
+
+    public string ClientInfo() =>
+        string.Join(
+                "\n",
+                counts.Select(kvp => $"** Workflow ID: {kvp.Key} {kvp.Value.ClientInfo()}"));
+
+    public ClientOutboundInterceptor InterceptClient(ClientOutboundInterceptor nextInterceptor) =>
+        new ClientOutbound(this, nextInterceptor);
+
+    public WorkflowInboundInterceptor InterceptWorkflow(WorkflowInboundInterceptor nextInterceptor) =>
+        new WorkflowInbound(this, nextInterceptor);
+
+    public ActivityInboundInterceptor InterceptActivity(ActivityInboundInterceptor nextInterceptor) =>
+        new ActivityInbound(this, nextInterceptor);
+
+    private sealed class ClientOutbound : ClientOutboundInterceptor
+    {
+        private MyCounterInterceptor root;
+
+        public ClientOutbound(MyCounterInterceptor root, ClientOutboundInterceptor next)
+            : base(next) => this.root = root;
+
+        public override Task<WorkflowHandle<TWorkflow, TResult>> StartWorkflowAsync<TWorkflow, TResult>(
+            StartWorkflowInput input)
+        {
+            var id = input.Options.Id ?? "None";
+            // Need to add an empty record if none exists
+            // we don't care if it doesn't add it as we will
+            // still increment the current value.
+            root.counts.TryAdd(id, new Counts());
+            Interlocked.Increment(ref root.counts[id].ClientExecutions);
+            return base.StartWorkflowAsync<TWorkflow, TResult>(input);
+        }
+
+        public override Task SignalWorkflowAsync(SignalWorkflowInput input)
+        {
+            var id = input.Id ?? "None";
+            Interlocked.Increment(ref root.counts[id].ClientSignals);
+            return base.SignalWorkflowAsync(input);
+        }
+
+        public override Task<TResult> QueryWorkflowAsync<TResult>(QueryWorkflowInput input)
+        {
+            var id = input.Id ?? "None";
+            Interlocked.Increment(ref root.counts[id].ClientQueries);
+            return base.QueryWorkflowAsync<TResult>(input);
+        }
+    }
+
+    private sealed class WorkflowInbound : WorkflowInboundInterceptor
+    {
+        private readonly MyCounterInterceptor root;
+
+        internal WorkflowInbound(MyCounterInterceptor root, WorkflowInboundInterceptor next)
+            : base(next) => this.root = root;
+
+        public override void Init(WorkflowOutboundInterceptor outbound)
+        {
+            base.Init(new WorkflowOutbound(root, outbound));
+        }
+
+        public override Task<object?> ExecuteWorkflowAsync(ExecuteWorkflowInput input)
+        {
+            var id = Workflow.Info.WorkflowId;
+            // Need to add an empty record if none exists
+            // we don't care if it doesn't add it as we will
+            // still increment the current value.
+            root.counts.TryAdd(id, new Counts());
+            Interlocked.Increment(ref root.counts[id].WorkflowExecutions);
```

Comment:
Note, this is not workflow executions, but workflow replays. You could get 1000 of these for the same workflow. Same for other counts in the workflow inbound/outbound. These will be double counted on every replay. You can consider not doing this on replay, or you can clarify in docs of the fields that they include replay attempts (you probably want the former).

Reply:
Oh wow. That's important to call out. 

I see that Workflow.Info has an Attempt field, but not sure it makes sense to use that to determine replay, as I think that is used for the number of retry attempts.

Reply:
There is `Workflow.Unsafe.IsReplaying` you can make sure is false before doing the increment (this is what we do in tracing interceptor for instance). It should be noted though that on task failure, this will run over and over in non-replaying situation, but besides task failure it will work.

Reply:
Fixed